lib/utils.sh

```diff
@@ -489,49 +489,36 @@ with_plugin_env() {
     local version="${version_info[0]}"
   fi
 
-  # create a new subshell to keep env
-  (
```

Comment:
Is this subshell not needed anymore?

Reply:
I am not exactly sure why it was needed in the first place. Any idea? 
The issue with the subshell is that, as the callback was called from the shell, there would always be an extra `bash` process running which kind of defeats the point of `exec`.

Reply:
I think it was used to contain the exported environment variables, but I'm not sure where they would leak if we didn't have it. So maybe it's not needed at all, but it was present prior to the change that remove `exec` I think.

Reply:
I understand that `exec-env` also exports env variables, which could "leak" but as asdf is executed in its own process, which will finally be replaced by the desired executable process, I do not really see why this would be an issue.
Maybe at some point in the past asdf was used as a function and not a command, so this was to avoid polluting shell environment?

Reply:
After a bit of digging I found that the subshell was introduced quite a while ago but there were not much details about why:
https://github.com/asdf-vm/asdf/commit/ab04ca9f559102f336dde513458816ed1fdb16bb#diff-969a0ff4a73b9d80109d01f5a51d7854
@HashNuke If you have a moment, it would be great if you could enlighten us :bulb:

Reply:
Good find. No idea what the original intention was, but I don't see any reason why we'd still need this. Worst case scenario it would just be polluting some asdf code right?

Reply:
> Worst case scenario it would just be polluting some asdf code right?

Yes, I think that this is all that could happen.